rs/ledger_suite/icrc1/ledger/src/lib.rs

```diff
@@ -883,3 +970,107 @@ impl<Tokens: TokensType> Ledger<Tokens> {
         }
     }
 }
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct StableAllowancesData {}
+
+impl AllowancesData for StableAllowancesData {
+    type AccountId = Account;
+    type Tokens = Tokens;
+
+    fn get_allowance(
+        &self,
+        account_spender: &(Self::AccountId, Self::AccountId),
+    ) -> Option<Allowance<Self::Tokens>> {
+        ALLOWANCES_MEMORY.with_borrow(|allowances| allowances.get(account_spender))
+    }
+
+    fn set_allowance(
+        &mut self,
+        account_spender: (Self::AccountId, Self::AccountId),
+        allowance: Allowance<Self::Tokens>,
+    ) {
+        ALLOWANCES_MEMORY
+            .with_borrow_mut(|allowances| allowances.insert(account_spender, allowance));
+    }
+
+    fn remove_allowance(&mut self, account_spender: &(Self::AccountId, Self::AccountId)) {
+        ALLOWANCES_MEMORY.with_borrow_mut(|allowances| allowances.remove(account_spender));
+    }
+
+    fn insert_expiry(
+        &mut self,
+        timestamp: TimeStamp,
+        account_spender: (Self::AccountId, Self::AccountId),
+    ) {
+        ALLOWANCES_EXPIRATIONS_MEMORY.with_borrow_mut(|expirations| {
+            expirations.insert((timestamp, account_spender), ());
+        });
+    }
+
+    fn remove_expiry(
+        &mut self,
+        timestamp: TimeStamp,
+        account_spender: (Self::AccountId, Self::AccountId),
+    ) {
+        ALLOWANCES_EXPIRATIONS_MEMORY.with_borrow_mut(|expirations| {
+            expirations.remove(&(timestamp, account_spender));
+        });
+    }
+
+    fn insert_arrival(
+        &mut self,
+        _timestamp: TimeStamp,
+        _account_spender: (Self::AccountId, Self::AccountId),
+    ) {
+    }
+
+    fn remove_arrival(
+        &mut self,
+        _timestamp: TimeStamp,
+        _account_spender: (Self::AccountId, Self::AccountId),
+    ) {
+    }
+
```

Comment:
Is the plan to remove these from the `AllowancesData` trait once the `HeapAllowancesData` struct is no longer needed?